fix(symfony): fix swagger config to support Symfony ConfigBuilders

alexndlm opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | 3.0
| Tickets       | n/a
| License       | MIT
| Doc PR       | n/a

Please see https://symfony.com/doc/current/configuration.html#using-php-configbuilders and https://symfony.com/doc/current/components/config/definition.html#array-node-options

alexndlm commented:
@soyuka would you please check this PR?

soyuka commented:
@vincentchalamon probably has a better idea then me of the impacts of this

vincentchalamon commented:
Hi @alexndlm,

Thanks for your PR. I've tested it in an API Platform project, but can't find any difference in the generated values comparing with the actual code, both dump me the following configuration:
```
array:1 [
  "foo" => array:2 [
    "name" => "Authorization"
    "type" => "header"
  ]
]
```

Can you explain why the use of `useAttributeAsKey` is necessary, and if possible add a test using it, please?

alexndlm commented:
@vincentchalamon, thanks that you check this PR.

I have updated the tests and rebased PR on 2.7.

This issue is only present with [Symfony PHP ConfigBuilders](https://symfony.com/doc/current/configuration.html#using-php-configbuilders).

Now Symfony generates the following Config Class:
<img width="1357" alt="Снимок экрана 2022-12-19 в 11 17 08" src="https://user-images.githubusercontent.com/6824784/208402822-61ec0424-3125-41b3-a448-f89c095aa6cc.png">

My API Platform config:
<img width="1037" alt="Снимок экрана 2022-12-19 в 11 20 34" src="https://user-images.githubusercontent.com/6824784/208403539-a9a6ad8f-b438-426e-8f2e-df82c7ddfb4e.png">


And it is not possible to specify the authorization name:
<img width="685" alt="Снимок экрана 2022-12-19 в 11 18 20" src="https://user-images.githubusercontent.com/6824784/208403067-220bfebd-7091-4325-b755-d2f42207100f.png">

With the fix in this PR, it will be the following:
<img width="1410" alt="Снимок экрана 2022-12-19 в 11 24 14" src="https://user-images.githubusercontent.com/6824784/208404344-4df42bbe-03c8-43d5-950f-f57ea3f1c591.png">

<img width="959" alt="Снимок экрана 2022-12-19 в 11 23 12" src="https://user-images.githubusercontent.com/6824784/208404382-f7114729-1dbe-4f6f-8ca3-316ee79599e0.png">

<img width="677" alt="Снимок экрана 2022-12-19 в 11 25 41" src="https://user-images.githubusercontent.com/6824784/208404503-d0e48579-5105-473f-9229-6e679835249c.png">

vincentchalamon commented:
Thanks for the clear clarification :smiley:
Can you add a test with ConfigBuilder to reveal the bug and fix it, please?

vincentchalamon commented on CHANGELOG.md:
```diff
@@ -1,5 +1,11 @@
+
+### Bug fixes
+
+* [4691](https://github.com/api-platform/core/pull/4691) Symfony: Fix swagger config to support Symfony ConfigBuilders  (#4691)
```
@soyuka isn't it auto-generated, or maybe do you have another process for it?

soyuka replied:
yes it is auto generated please remove theses changes

alexndlm replied:
Done

vincentchalamon reviewed (approved):
Bad commitlint (easy to fix), otherwise :+1:
Good job @alexndlm!

vincentchalamon commented:
Good job @alexndlm, thanks!
